lib/offline/OfflineStorage_Room.cpp

```diff
@@ -82,9 +76,17 @@ namespace MAT_NS_BEGIN
     void
     OfflineStorage_Room::ConnectedEnv::pushLocalFrame(uint32_t frameSize)
     {
-        if (env->PushLocalFrame(frameSize) == 0)
-        {
-            ++push_count;
+        try {
+            if (env->PushLocalFrame(frameSize) == JNI_OK) {
+                ++push_count;
+            }
+        }
+        catch (std::exception e) {
```

Comment:
Catch by const&, C++17 mitigates this with Guaranteed Copy Elision, which allows copy free catch by value. As the SDK only requires C++11, we should adhere to best practices for that language version.

Reply:
Applies to all catch blocks :)

Reply:
Also, since std::exception is a base class, we may get into some slicing problems if we catch by value here :)